File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(int[] elements) {
        HashSet<int> set = new HashSet<int>();
        int n = elements.Length;
        
        for(int i = 0; i < n; i++)
        {
            int sum = 0;
            for(int j = 0; j < n; j++)
            {
                sum += elements[(i + j) % n];
                set.Add(sum);
            }
        }
        return set.Count;
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> elements) {
    set<int> sums;
    int n = elements.size();
    vector<int> arr = elements;
    
    for(int i = 0; i < n; i++)
    {
        arr.push_back(elements[i]);
    }
    
    for(int i = 1; i <= n; i++)
    {
        int sum = 0;
        for(int j = 0; j < i; j++)
        {
            sum += arr[j];
        }
        sums.insert(sum);
        
        for(int j = 1; j < n; j++)
        {
            sum -= arr[j - 1];
            sum += arr[j + i - 1];
            sums.insert(sum);
        }
    }
    return sums.size();
}